core/src/data-generator/batch-replayer.ts

```diff
@@ -0,0 +1,185 @@
+ * 
+ * 2. resources/lambdas/write-in-batch
+ * Take a file path, filter only records within given time range, adjust the the time with offset to
+ * make it looks like just being generated. Then write the output to the `sinkBucket`
+ */
+export class BatchReplayer extends Construct {
+
+  /**
+   * Dataset used for replay
+   */
+  public readonly dataset: PartitionedDataset;
+
+  /**
+   * Frequency (in Seconds) of the replaying. The batch job will start
+   * for every given frequency and replay the data in that period
+   */
+  public readonly frequency: number;
+
+  /**
+   * Sink bucket where the batch replayer will put data in
+   */
+  public readonly sinkBucket: Bucket;
+
+  /**
+   * Maximum file size for each output file. If the output batch file is,
+   * larger than that, it will be splitted into multiple files that fit this size.
+   * 
+   * Default to 100MB (max value)
+   */
+  public readonly outputFileMaxSizeInBytes?: number;
+
+  constructor(scope: Construct, id: string, props: BatchReplayerProps) {
+    super(scope, id);
+
+    this.dataset = props.dataset;
+    this.frequency = props.frequency || 60;
+    this.sinkBucket = props.sinkBucket;
+    this.outputFileMaxSizeInBytes = props.outputFileMaxSizeInBytes || 100 * 1024 * 1024; //Default to 100 MB
+    
+
+    /**
+     * Find all paths within the time range from the manifest file
+     */
+    const findFilePathsFn = new lambda.DockerImageFunction(this, "findFilePathFn", {
+      memorySize: 1024,
+      code: lambda.DockerImageCode.fromImageAsset(path.join(__dirname, '../../lib/data-generator/resources/lambdas/find-file-paths')),
+      logRetention: RetentionDays.ONE_DAY,
+      timeout: Duration.minutes(15),
+    });
+
+    // Grant access to read S3
+    const { bucketName, objectKey } = this.dataset.manifestLocation;
+    findFilePathsFn.role?.attachInlinePolicy(
+      new Policy(this, 'read-manifest-file-policy', {
+        statements: [
+          new PolicyStatement({
+            actions: ['s3:GetObject'],
+            resources: [
+              `arn:aws:s3:::${bucketName}/${objectKey}`
+            ],
+          }),
+        ],
+      }),
+    );
+
+    const findFilePathsFnTask = new LambdaInvoke(this, "findFilePathFnTask", {
+      lambdaFunction: findFilePathsFn,
+      payload: TaskInput.fromObject({
+        frequency: props.frequency,
+        manifestFileBucket: this.dataset.manifestLocation?.bucketName,
+        manifestFileKey: this.dataset.manifestLocation?.objectKey,
+        triggerTime: JsonPath.stringAt('$$.Execution.Input.time'),
+        offset: '' + this.dataset.offset,
+      }),
+      // Retry on 500 error on invocation with an interval of 2 sec with back-off rate 2, for 6 times
+      retryOnServiceExceptions: true,
+      outputPath: "$.Payload",
+    });
+
+
+    /**
+     * Write data in batch step
+     */
+    const writeInBatchFn = new lambda.DockerImageFunction(this, "writeInBatchFn", {
```

Comment:
can we use PreBundledFunction?

Reply:
No, it exceeds Lambda package size. I was considering a Lambda layer but there is no official one.